### PR TITLE
Modifications to support OnDemand app development tutorial with Jupyter 

### PR DESCRIPTION
**Pull request by ericfranz**
- add jupyter to compute node including required libs for example batch job
- add debug partition for demonstrating enabling specifying the partition in an app
- make hpcadmin an OnDemand app developer

### CODE REVIEW

File: ondemand/install.sh [hunk 1]
```diff
@@ -99,3 +99,11 @@ log_info "Generating new httpd24 and dex configs.."
...
 yum clean all
 rm -rf /var/cache/yum
+
+log_info "Cloning repos to assist with app development.."
+mkdir -p /var/git
+git clone https://github.com/OSC/bc_example_jupyter.git --bare /var/git/bc_example_jupyter
+
+log_info "Enabling app development for hpcadmin..."
+mkdir -p /var/www/ood/apps/dev/hpcadmin
+ln -s /home/hpcadmin/ondemand/dev /var/www/ood/apps/dev/hpcadmin/gateway
```
Comment (johrstrom) [inline ondemand/install.sh:109]: I think we also need a `su hpcadmin -c "mkdir -p /home/hpcadmin/ondemand/dev"` right?

Comment (ericfranz) [reply ondemand/install.sh:109]: The dashboard creates this automatically if it doesn't exist. I was able to login as the hpcadmin user and access the sandbox apps and it worked.

Comment (aebruno) [reply ondemand/install.sh:109]: Just did a quick test and looks like it's created ok:

```[hpcadmin@ondemand ~]$ ls -alh /home/hpcadmin/ondemand/dev
total 8.0K
drwxr-xr-x 2 hpcadmin hpcadmin 4.0K Jul 15 19:46 .
drwxr-xr-x 4 hpcadmin hpcadmin 4.0K Jul 15 19:46 ..
```

**Comment by aebruno** [comment]
This looks good to me. Just quick question, should I be seeing any apps under "My Sandbox Apps (Development)"? Currently the list is empty.

sandbox

**Comment by ericfranz** [comment]
@aebruno no apps yet. The tutorial starts with instructing users to create a new app by inserting the path to the git repo (instead of the URL since we embed the git repo in the image).

**Pull request merged**